src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -131,24 +134,23 @@ public static void ValidateDatabaseName(string databaseName)
             }
         }
 
-        public static SqlConnection GetConnectionIfDatabaseNotExists(string connectionString, string databaseName)
+        public static bool DoesDatabaseExist(SqlConnection masterDbConnection, string databaseName)
         {
-            // Connect to master database to evaluate if the requested database exists
-            var masterConnectionBuilder = new SqlConnectionStringBuilder(connectionString) { InitialCatalog = string.Empty };
-            var connection = new SqlConnection(masterConnectionBuilder.ToString());
-            connection.Open();
+            EnsureArg.IsNotNull(masterDbConnection, nameof(masterDbConnection));
+
+            masterDbConnection.Open();
```

Comment:
You probably need to check if the connection is already open here since I believe it will throw an exception if you try to open it again.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnection.open?view=sqlclient-dotnet-core-2.0#Microsoft_Data_SqlClient_SqlConnection_Open

Reply:
Makes sense. Will update.